kubectl-deploy/bin/kubectl-deploy

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+# Blue/Green deployment kubectl plugin to safely release a new version of app
+# on k8s environment included LB, ingressroute, service and deployment
+#                                                   ┌────────────────┐
+#                                            ┌─────►│Blue deployment │
+# ┌─────────┐   ┌─────────────┐   ┌────────┐ │      └────────────────┘
+# │   LB    ├──►│ingressroute ├──►│  SVC   ├─┤
+# └─────────┘   └─────────────┘   └────────┘ │      ┌────────────────┐
+#                                            └─────►│Green deployment│
+#                                                   └────────────────┘
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+DEFAULT_NAMESPACE="default"
+DEFAULT_TAG="latest"
+DEFAULT_TIMEOUT=120
+API_SERVER="127.0.0.1:8443"
+CERT="ca.crt"
+DOCKER_REPO="docker.io/phantasm"
```

Comment:
We shouldn't be using default here. Instead it is better to fail without proceeding any of the steps if the value is not present.

Reply:
Removed optional parameters.